app/assets/javascripts/components/buttons/generic-button-component.js

```diff
@@ -0,0 +1,41 @@
+ManageIQ.angular.app.component('genericButtonComponent', {
+  bindings: {
+    angularForm: '=',
+    modelCopy: '=',
+    model: '=',
+    redirectUrl: '=?',
+    newRecord: '<',
+    entity: '=?',
+    entityName: '=?',
+    onSaveClicked: '&?',
+    onAddClicked: '&?',
+  },
+  controllerAs: 'vm',
+  controller: genericButtonController,
+  templateUrl: '/static/buttons/generic_button_component.html.haml',
+});
+
+genericButtonController.$inject = ['miqService', '$window'];
+
+function genericButtonController(miqService) {
+  var vm = this;
+
+  vm.$onInit = function() {
+    vm.saveable = miqService.saveable;
+
+    vm.resetClicked = function() {
```

Comment:
I fully expect to see multiple use cases for buttons.

I have captured the most common use case that we hit at least 75% to 85% of the times in this PR.
One of the main goals that I had in mind, (I will repeat), is to address the CC error `Similar code found in 'n' other places`. 
In fact, as we start using components more and more, we should completely eliminate all common code, convert it to a component and re-use that component.

So it's really not about if I want to insist on something or not, but it's about making things better by removing redundancy.